### PR TITLE
DOC/MAINT: simplify release entries

### LINKED ISSUE #17710: MAINT, DOC: doc_requirements and make dist and pooch
This patch is needed to succeed with `make dist` locally during the release process, to build the docs:

```diff
diff --git a/doc_requirements.txt b/doc_requirements.txt
index f601052f3..8f66e5886 100644
--- a/doc_requirements.txt
+++ b/doc_requirements.txt
@@ -5,3 +5,4 @@ pydata-sphinx-theme==0.9.0
 sphinx-design>=0.2.0
 numpydoc
 matplotlib>2
+pooch
```

It isn't caught by our CI because CircleCI installs `pooch` after the `pip install -r doc_requirements.txt` step. This really only matters for release management I think, but it may be helpful to pack the true requirements into that file.

### PR DESCRIPTION
**Pull request by tupui**
Follows NumPy's doc. This greatly simplify the tree structure. I think it's simpler and clearer to make releases then.

Instead of having to add 2 files for the release entry, only a single file is needed and the link in `release.rst` directly points to it instead of an empty file which only does the linking.

The added bonus is that for folks working on the doc it's cleaner.

Also closes #17710 by adding Pooch to the doc requirements.

### CODE REVIEW

File: doc/source/release.rst [hunk 1]
```diff
@@ -2,68 +2,68 @@
 Release Notes
 *************
 
-This is the list of changes to SciPy between each release. For full details,
+This is the list of changes to SciPy between each  For full details,-notes
```
Comment (j-bowhay) [inline doc/source/release.rst:5]: Unintentional deletion of release?
```suggestion
This is the list of changes to SciPy between each release. For full details,-notes
```

Comment (tupui) [reply doc/source/release.rst:5]: Oops yes thanks! Let's put this in when we are done.

**Comment by rgommers** [comment]
Thanks @tupui. If this goes in, the relevant commit should probably be added to `git-blame-ignore-revs` afterwards. I'm fine with this change - I'll defer review/decision to @tylerjereddy.

**Review by tylerjereddy — COMMENTED** [review]
FWIW, it doesn't really make it that much easier for me, and indeed I'm more likely to mess it up because of habit and just copy/pasting from old PRs/release workflows, which is what I usually do.

That said, I don't really mind the change, so I guess I'm saying that if this makes life easier for the docs folks I'm not going to block it.

**Review by melissawm — APPROVED** [review]
Thank you! This does indeed make it a lot easier 😄

**Comment by tupui** [comment]
Alright thanks for the feedbacks. Based on that I will merge then.

**Comment by tupui** [comment]
I will make the followup to add the commit to the ignore list.

**Pull request merged**